### PR TITLE
Added enum isLight, isDark and isSystem getter to ThemeMode enum

### PR DESCRIPTION
**Pull request by stan-at-work**
Added the following enums to the ThemeMode enum:

`isLight` => quick way to check if enum is ThemeMode.light.
`isDark` => quick way to check if enum is ThemeMode.dark.
`isSystem` => quick way to check if enum is ThemeMode.system.

*List which issues are fixed by this PR. You must list at least one issue.*
https://github.com/flutter/flutter/issues/159569

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

### CODE REVIEW

**Review by nate-thegrate — COMMENTED** [review]
It's great to see the enhanced enum features here!

Let me know what you think about the `isLight`/`isDark` fields.

File: packages/flutter/lib/src/material/app.dart [hunk 1]
```diff
@@ -55,15 +55,36 @@ const TextStyle _errorTextStyle = TextStyle(
...
+
+  /// Checks if the themeMode is [ThemeMode.light].
+  ///
+  /// {@macro flutter.material.ThemeMode.themeMode.light}
+  bool get isLight => this == ThemeMode.light;
+
+  /// Checks if the themeMode is [ThemeMode.dark].
+  ///
+  /// {@macro flutter.material.ThemeMode.themeMode.dark}
+  bool get isDark => this == ThemeMode.dark;
```
[the remark this answers is not in the document]
Comment (stan-at-work) [reply packages/flutter/lib/src/material/app.dart:85]: This is a good idea.

**Review by Piinks — COMMENTED** [review]
Hey @stan-at-work welcome! Thank you for contributing! This will also need tests to verify it works as expected. Thanks!

**Comment by stan-at-work** [comment]
> Hey @stan-at-work welcome! Thank you for contributing! This will also need tests to verify it works as expected. Thanks!

They are added

File: packages/flutter_test/test/theme_mode_test.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    expect(dark.isSystem, false);
+
+    expect(light.isDark, false);
+    expect(light.isLight, true);
+    expect(light.isSystem, false);
+
+    expect(system.isDark, false);
+    expect(system.isLight, false);
+    expect(system.isSystem, true);
+  });
```
Comment (nate-thegrate) [inline packages/flutter_test/test/theme_mode_test.dart:33]: This is hilarious 😂 (and also very well done!)

**Review by nate-thegrate — CHANGES_REQUESTED** [review]
I think implementing [enhanced enum](https://dart.dev/language/enums#declaring-enhanced-enums) features here would be a missed opportunity if the `isDark` field doesn't accurately reflect whether the current theme brightness is dark.

I think it'd be easiest to mark this guy as a draft until https://github.com/flutter/engine/pull/56841 lands. @stan-at-work how does that sound?

**Comment by stan-at-work** [comment]
> I think implementing [enhanced enum](https://dart.dev/language/enums#declaring-enhanced-enums) features here would be a missed opportunity if the `isDark` field doesn't accurately reflect whether the current theme brightness is dark.
> 
> I think it'd be easiest to mark this guy as a draft until [flutter/engine#56841](https://github.com/flutter/engine/pull/56841) lands. @stan-at-work how does that sound?

What do you mean by this exacly ?

**Comment by stan-at-work** [comment]
> I think implementing [enhanced enum](https://dart.dev/language/enums#declaring-enhanced-enums) features here would be a missed opportunity if the `isDark` field doesn't accurately reflect whether the current theme brightness is dark.
> 
> I think it'd be easiest to mark this guy as a draft until [flutter/engine#56841](https://github.com/flutter/engine/pull/56841) lands. @stan-at-work how does that sound?

I'm creating the test right now. will be pushing them when ready

**Comment by nate-thegrate** [comment]
> > I think implementing [enhanced enum](https://dart.dev/language/enums#declaring-enhanced-enums) features here would be a missed opportunity if the `isDark` field doesn't accurately reflect whether the current theme brightness is dark.
>
> What do you mean by this exacly?

I would expect `isDark` to be true if we're using `ThemeMode.system` and the system brightness is dark. It'd be great if we could replace the current getter with something similar to the idea from https://github.com/flutter/flutter/pull/159570#discussion_r1868589981.

**Comment by matanlurey** [comment]
Similar to https://github.com/flutter/engine/pull/56841#issuecomment-2521751434 I'm not supportive of this change, and have identical advice to what is provided in that comment.

**Review by matanlurey — CHANGES_REQUESTED** [review]
No LGTM; respective feature proposal was never reviewed prior to this PR. Leaving open to see if other members of the team feel strongly otherwise.

**Comment by matanlurey** [comment]
Closing for same reason as https://github.com/flutter/engine/pull/56841#issuecomment-2523753544.

**Comment by nate-thegrate** [comment]
Sounds good! I'm down with closing this one and exploring other ways to improve the related APIs.

However: based on the interactions in [a previous Code of Conduct PR](https://github.com/flutter/flutter/pull/158195), I feel entitled to share my thoughts at this point, despite how they won't affect the outcome here.

<br><br>

> respective feature proposal was never reviewed prior to this PR.

Whether or not the proposal was reviewed prior to this PR is inconsequential, given that I reviewed these changes and was decidedly in support of expanding on the enums.

> I prefer that we keep dart:ui itself as "minimal" as possible.

This is a reasonable preference, though it only applies to the engine PR, not this one.

> If you want to define convenience getters in your own projects/packages that is of course possible via extension methods.

Unfortunately the style guide prohibits use of extension methods.

<br>

There are several alternative ways to improve the existing "theme brightness" APIs, and I also feel that the style guide's advice regarding `extension` ought to be revisited.

But I'd like to mention a few things from Flutter's [code of conduct](https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md):

- When multiple contributors disagree on the direction for a particular patch or the general direction of the project, the conflict should be resolved by communication.
- A conflict is addressed when the participants agree that the final solution is _better_ than all the conflicting proposals.
- Listen. Consider and acknowledge people's points before responding.

<br>

The opinion of a more senior project member is especially valuable, but it's not a substitute for open discussion. I made a few points in [a previous comment](https://github.com/flutter/engine/pull/56841#issuecomment-2521932495), and there's no indication that any of them were acknowledged before the pull requests were closed:

- Having null safety enforced by static analysis makes it easier to catch a potentially `null` value.
- Using `.isDark(context)` ensures that a MediaQuery dependency is only created if `ThemeMode.system` is selected, and incorporating this logic into the enum makes it easy for this type of performance tweak to apply everywhere.

<br>

In the grand scheme of things, "adding enum fields" isn't anywhere near as important as "giving every contributor a chance to be heard". I will continue making effort to promote a spirit of openness and mutual respect in this community, and I expect other team members to do the same.

**Comment by Piinks** [comment]
> Unfortunately the style guide prohibits use of extension methods.

FWIW, the style guide is specific to the flutter repo, not other users' projects.

**Comment by nate-thegrate** [comment]
True that.

My intent was for the benefit to apply to the framework as well, per https://github.com/flutter/engine/pull/56841#issuecomment-2521932495:

> I created [a draft PR](https://github.com/nate-thegrate/flutter/pull/10/files) to show how this API could be utilized in Flutter's codebase.

**Comment by dabblingwithcode** [comment]
What does fwiw mean?

**Pull request closed without merging**